internal/resources/sourcesecret/scope/scope.go

```diff
@@ -0,0 +1,93 @@
+/*
+Copyright © 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package scope
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"golang.org/x/exp/slices"
+
+	sourcesecretlustermodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/sourcesecret/cluster"
+	sourcesecretlustergroupmodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/sourcesecret/clustergroup"
+	commonscope "github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/common/scope"
+)
+
+// ScopedFullname is a struct for all types of kustomization full names.
```

Comment:
I highly doubt kustomization and sourcesecret names are used interchangeably ? @Shubbhang351 correct me if I am wrong

Reply:
kustomization and sourcesecret are different I have updated this all the places, PTAL